arangod/Transaction/Manager.cpp

```diff
@@ -120,9 +121,12 @@ void Manager::unregisterFailedTransactions(std::unordered_set<TRI_voc_tid_t> con
 
 void Manager::registerTransaction(TRI_voc_tid_t transactionId,
                                   std::unique_ptr<TransactionData> data,
-                                  bool isReadOnlyTransaction) {
-  if (!isReadOnlyTransaction && !isFollowerTransactionId(transactionId)) {
+                                  bool isReadOnlyTransaction,
+                                  bool isFollowerTransaction) {
+  if (!isReadOnlyTransaction && !isFollowerTransaction) {
+    LOG_TOPIC("ccdea", TRACE, Logger::TRANSACTIONS) << "Acquiring read lock for tid " << transactionId;
     _rwLock.readLock();
+    LOG_TOPIC("ccdeb", TRACE, Logger::TRANSACTIONS) << "Got read lock for tid " << transactionId << " nrReadLocked: " << ++_nrReadLocked;
```

Comment:
The counter `_nrReadLocked` is only increased when the log level is set to trace. Otherwise it will stay at zero. It has no side effects though.
In addition, if one changes the log level at runtime, it is possible that the counter is not increased (log level > trace) but decreased (log level == trace). In this case, the counter would underflow. Again, no side effects apart from misleading log messages, but it isn't ideal.

Reply:
Good catch! This is an oversight and is fixed now.